tests/protocol/test_wire.py

```diff
@@ -1114,15 +1048,170 @@ def host_func(*args):
 
         # Assert we've called both the direct channel function (once) and the host channel function (twice).
         # After the host channel succeeds, the host plugin should have been set as the default channel.
-        with patch('azurelinuxagent.common.protocol.wire.WireClient.update_goal_state') as mock_update_goal_state:
+        with patch('azurelinuxagent.common.protocol.wire.WireClient.update_host_plugin_from_goal_state') as mock_update_host_plugin_from_goal_state:
             ret = client.send_request_using_appropriate_channel(direct_func, host_func)
             self.assertEquals(42, ret)
             self.assertEquals(1, direct_func.counter)
             self.assertEquals(2, host_func.counter)
-            self.assertEquals(1, mock_update_goal_state.call_count)
+            self.assertEquals(1, mock_update_host_plugin_from_goal_state.call_count)
             self.assertEquals(True, client.get_host_plugin().is_default_channel())
 
 
+class GoalStateTestCase(AgentTestCase):
+    """
+    Base class for UpdateGoalStateTestCase and UpdateHostPluginFromGoalStateTestCase
+    """
+    def setUp(self):
+        AgentTestCase.setUp(self)
+
+        self.mock_data = mockwiredata.WireProtocolData(mockwiredata.DATA_FILE)
+        self.mock_http_get = patch("azurelinuxagent.common.utils.restutil.http_get", side_effect=self.mock_data.mock_http_get)
+        self.mock_crypt_util = patch("azurelinuxagent.common.protocol.wire.CryptUtil", side_effect=self.mock_data.mock_crypt_util)
+        self.mock_http_get.start()
+        self.mock_crypt_util.start()
+
+        # Use the WireProtocol to create a test WireClient.
+        # WireProtocol.detect() creates the transport certs needed to retrieve the goal state and also sets the initial goal state.
+        # WireClient.get_host_plugin() forces the host plugin to be instantiated
+        protocol = WireProtocol(WIRESERVER_URL)
+        protocol.detect()
+        self.wire_client = protocol.client
+        self.wire_client.get_host_plugin()
+
+    def tearDown(self):
+        self.mock_crypt_util.stop()
+        self.mock_http_get.stop()
+        AgentTestCase.tearDown(self)
+
+
+class UpdateGoalStateTestCase(GoalStateTestCase):
+    """
+    Tests for WireClient.update_host_plugin_from_goal_state()
```

Comment:
NIT: Typo - should be 
`Tests for WireClient.update_goal_state()`

Reply:
thanks